src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -679,19 +679,27 @@ function _linear_map_hrep(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
                           algo::LinearMapInverse) where {N}
     inverse = algo.inverse
     constraints_P = constraints_list(P)
-    constraints_MP = similar(constraints_P)
+    constraints_MP = _similar(constraints_P)
     @inbounds for (i, c) in enumerate(constraints_P)
         cinv = vec(_At_mul_B(c.a, inverse))
         constraints_MP[i] = LinearConstraint(cinv, c.b)
     end
     return constraints_MP
 end
 
+function _similar(constraints::Vector{LinearConstraint{N, VN}}) where {N, VN<:AbstractVector{N}}
+    return Vector{LinearConstraint{N, Vector{N}}}(undef, length(constraints))
+end
+
+function _similar(constraints::LinearConstraint{N, VN}) where {N, VN<:AbstractVector{N}}
+    return LinearConstraint{N, Vector{N}}()
+end
```

Comment:
We do have to continue by adding more dispatches on similar types; for example, handle the static array case, `reinterpret`ing vectors after the loop etc. Note that the type of normal vector is not in general preserved after you transform it. For example, consider that `P` is axis aligned, the constraints are represented  minimally, ie. with single entry vector. But after the transformation, the normal vectors cannot longer be represented by such type in general.

Reply:
> The second method is not used anywhere.

The second method is used when this function receive a halfpsace. We have a test that takes this codepath.

Reply:
Can we at least rename the function? It has nothing to do with the `similar` function in Julia which gives you the *same* type.

```julia
help?> similar

Create an uninitialized mutable array with the given element type and size,
based upon the given source array.
```

> > The second method is not used anywhere.
> 
> The second method is used when this function receives a halfspace. We have a test that takes this codepath.

If you look at the changed files in this PR and search for `_similar`, you will find that it is only applied to results of `constraints_list`, which is always a vector of constraints.

Furthermore, the code does not make sense. You cannot create a `LinearConstraint` without any arguments. This alone proves that this method is never called in the tests.

> Note that the type of normal vector is not in general preserved after you transform it.

I understand that.

Reply:
> You cannot create a LinearConstraint without any arguments.

True. I must have made some intermediate change and forgot to update this part, but this function can be safely removed. Since the other cases are not implemented yet, i will just hardcode the vector.